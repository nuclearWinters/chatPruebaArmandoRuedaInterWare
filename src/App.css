html, body {
  font-family: Roboto;
  font-weight: 400;
}

.app {
  height: 100vh;
  display: flex;
  flex-direction: column;
}

.header {
  z-index: 300;
  background-color: #49a9ea;
  display: flex;
  flex-direction: row;
  height: 36px;
  align-items: center;
  justify-content: center;
}

.header .tab {
  font-size: 18; 
  border-radius: 5;
  margin: 0px 5px; 
  padding: 3px 15px; 
  color: white; 
  background-color: rgba(255,255,255,.1)
}

.header .first {
  margin-left: 20px;
}

.header .title {
  flex: 1;
  text-align: center;
  color: white;
  font-size: 14px;
}

.header .iconBox {
  background-color: rgba(255,255,255,0.1); 
  margin: 0px 10px; 
  height: 26px; 
  width: 26px; 
  align-items: center;
  justify-content: center; 
  display: flex; 
  border-radius: 52px
}

.header .imageProfile {
  border-radius: 48px; 
  margin-left: 10px; 
  margin-right: 25px;
}

::-webkit-input-placeholder {
  color: #797979;
}

