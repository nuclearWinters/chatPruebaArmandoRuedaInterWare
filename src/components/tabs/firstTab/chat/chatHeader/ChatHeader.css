.chatHeader {
    height: 48px;
    background-color: #e6e6e6; 
    display: flex; 
    flex-direction: row;
}

.chatHeader .image {
    border-radius: 48px; 
    margin: auto 0px auto 60px;
}

.chatHeader .profileName {
    margin-left: 16px; 
    margin: auto 0px auto 16px; 
}

.chatHeader .profileName .profileStatus {
    display: flex; 
    flex-direction: row; 
    align-items: center;
}

.chatHeader .profileName .profileStatus .icon {
    height: 13px; 
    width: 13px; 
    display: flex; 
    align-items: center; 
    justify-content: center;
}

.chatHeader .profileName .profileStatus .circle {
    background-color: #22b573;
    height: 13px; 
    width: 13px; 
    border-radius: 26px;
    margin-left: 20px;
}

.chatHeader .profileName .profileStatus .status {
    margin-right: 20px;
}

.chatHeader .profileName .profileStatus .option {
    width: 130px;
}

.chatFont {
    color: #797979; 
    margin-left: 10px; 
    font-size: 14px; 
    font-weight: 300;
}

.iconChatHeader {
    height: 16px; 
    width: 16px; 
    margin: auto 0px auto 0px
}

.inputChatHeader {
    height: 30px; 
    display: flex; 
    border-radius: 4px;
    margin: auto;
    flex: 1;
    align-items: flex-end;
    justify-content: flex-end;
}

.inputChatHeaderBoxLast {
    display: flex; 
    flex-direction: row; 
    align-items: center; 
    justify-content: center; 
    margin-right: 30px;
}

.inputChatHeaderBox {
    display: flex; 
    flex-direction: row; 
    align-items: center; 
    justify-content: center; 
    margin-right: 15px;
}

.selected {
    color: #49a9ea
}

.iconSelected {
    height: 16px; 
    width: 16px
}

.chatInput {
    font-family: Roboto; 
    font-weight: 300; 
    font-size: 16px;
    outline: none;
    height: 25px;
    width: 160px;
    border-top-left-radius: 5px;
    border-bottom-left-radius: 5px; 
    background-color: #ffffff; 
    border-width: 0px;
    margin: auto 0px;
    padding-left: 10px;
}

.iconSearch {
    background-color: #ffffff; 
    height: 27px; width: 16px; margin: auto 0px;
    padding-right: 5px;
    border-top-right-radius: 5px;
    border-bottom-right-radius: 5px; 
}

.inputChatHeaderBoxFirst {
    display: flex; 
    margin-right: 15px; 
    margin-left: 25px; 
    align-items: center; 
    justify-content: center
}

.inputChatHeaderBoxSelected {
    padding: 0px 5px; 
    border-bottom: 2px solid #49a9ea; 
    flex-direction: row; 
    align-items: center; 
    justify-content: center; 
    display: flex;
}

.menu.open {
    transform: translateX(0);
}

.burgerIcon {
    height: 48px; 
    margin-right: 30px;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    display: none;
}

.burgerIcon div {
    width: 28px; 
    background-color: #797979; 
    height: 3px; 
    margin: 3px 0px;
}

@media only screen and (max-width: 1120px) {
    .inputChatHeaderBoxSelected .chatFont {
        display: none;
    }
    .inputChatHeaderBox .chatFont {
        display: none;
    }
    .inputChatHeaderBoxLast .chatFont {
        display: none;
    }
}

@media only screen and (max-width: 970px) {
    .profileStatus .status {
        display: none;
    }
    .profileStatus .option {
        display: none;
    }
    .profileStatus .circle {
        margin-right: 20px;
        margin-left: 0px !important
    }
    .profileStatus svg {
        margin-right: 20px;
    }
}

@media only screen and (max-width: 800px) {
    .menu {
        box-shadow: 1px 0px 7px rgba(0, 0, 0, 0.5);
        position: fixed;
        top: 36px;
        left: 0px;
        bottom: 0px;
        z-index: 200;
        transform: translateX(-100%);
        transition: transform 0.3s ease-out;
    }
    .inputChatHeaderBoxLast {
        margin-right: 15px;
    }
    .burgerIcon {
        display: flex;
    }
}

@media only screen and (max-width: 580px) {
    .inputChatHeaderBoxLast {
        margin-right: 15px;
    }
    .burgerIcon {
        display: flex;
    }
    /*.chatHeader .image {
        margin-left: 15px; 
    }*/
    .chatInput {
        display: none
    }
    .iconSearch {
        display: none
    }
}

@media only screen and (max-width: 470px) {
    .chatHeader .image {
        margin-left: 15px; 
    }
}
