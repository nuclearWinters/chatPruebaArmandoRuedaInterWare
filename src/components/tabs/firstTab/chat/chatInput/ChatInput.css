.chatInputFixed {
    height: 70px; 
    background-color: #f2f2f2; 
    display: flex;
}

.imageChatInputFixed {
    border-radius: 48px; margin-left: 60px; margin: auto 0px auto 60px;
}

.inputOfChatFixedBox {
    min-width: 60px;
    background-color: #ffffff; 
    height: 30px; 
    display: flex; 
    border-radius: 4px; 
    margin: auto 0px auto 20px; 
    flex: 1;
    /*min-width: 180;*/
}

.inputOfChatFixed {
    font-family: Roboto; 
    font-weight: 300; 
    font-size: 16px; 
    outline: none; 
    flex: 1; 
    height: 25px;
    border-radius: 5px; 
    background-color: rgba(0,0,0,0);
    border-width: 0px; 
    margin: auto 0px auto 10px
}

.iconChatInputFixedFirst {
    height: 16px; 
    width: 16px; 
    margin: auto 0px auto 30px;
}

.iconChatInputFixed {
    height: 16px; 
    width: 16px; 
    margin: auto 0px auto 20px;
}

.iconChatInputFixedLast {
    height: 16px; 
    width: 16px; 
    margin: auto 60px auto 20px;
}

@media only screen and (max-width: 470px) {
    .imageChatInputFixed {
        margin-left: 15px !important; 
    }
}

@media only screen and (max-width: 430px) {
    .imageChatInputFixed {
        margin-left: 15px !important; 
    }
    .iconChatInputFixedLast {
        margin-right: 20px !important;
    }
}

@media only screen and (max-width: 400px) {
    .imageChatInputFixed {
        margin-left: 15px !important; 
    }
    .iconChatInputFixedLast {
        margin-right: 20px !important;
    }
    .iconChatInputFixedFirst {
        display: none
    }
    .iconChatInputFixed {
        display: none
    }
    .iconChatInputFixedLast {
        display: none
    }
    .inputOfChatFixedBox {
        margin-right: 20px !important;
    }
}