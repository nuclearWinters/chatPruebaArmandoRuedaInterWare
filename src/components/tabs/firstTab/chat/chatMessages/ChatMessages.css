.chatMessages {
    overflow-y: scroll; 
    flex: 1;
}

.messageBox {
    display: flex; flex-direction: column
}

.messageData {
    margin-left: 90px; 
    margin-top: 0px; 
    display: flex; 
    flex-direction: row; 
    align-items: center;
}

.messageMessage {
    margin-left: 133px; 
    font-size: 15px; 
    color: black; 
    font-weight: 300;
    padding-bottom: 20px
}

.messageDate {
    margin-left: 15px; 
    font-size: 14px; 
    color: #797979;
}

.messageUser {
    margin-left: 15px; font-size: 16px; color: black;
}

.messageImage {
    border-radius: 48px
}

.dividerLine {
    border: 1px solid #cccccc
}

.dividerDate {
    text-align: center;
    padding-top: 10px; 
    font-size: 15px;
    color: black; 
    font-weight: 300;
    padding-bottom: 20px
}

.chatMessagePaddingBox {
    padding-top: 25px; 
    padding-bottom: 10px
}

@media only screen and (max-width: 430px) {
    .messageData {
        margin-left: 20px; 
    }
    
    .messageMessage {
        margin-left: 63px; 
    }
}