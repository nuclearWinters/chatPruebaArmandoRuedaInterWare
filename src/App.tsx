import React from 'react';
import './App.css';
import MainClass from "./components/tabs/firstTab/Main"
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faUsers, faRocket, faTrophy } from '@fortawesome/free-solid-svg-icons'
import yo from "./imgs/CARLOS.png"
import { Provider } from "react-redux"
import store from "./store"

const App: React.FC = () => {
  return (
    <Provider store={store}>
      <div className="app">
        <div className="header">
          <div className="tab first">{`(◕‿◕)♡`}</div>
          <div className="tab">{`<3 (ﾉ◕ヮ◕)ﾉ`}</div>
          <div className="title">
            <div className="entry">
              <div className="entry__inner">
                  <span className="entry__text">{`You are my god, I couldn't possibly ever thank you enough. <3  I couldn't possibly think of a good enough reward for the very best Patron I will ever have. <3 If you achieve this tier, please private message me. I'm not sure what reward I could give you, but we'll talk about it. I love you.`}</span>
              </div>
            </div>
          </div>
          <div className="iconBox"><FontAwesomeIcon color="#f7f8f9" icon={faRocket} /></div>
          <div className="iconBox"><FontAwesomeIcon color="#f7f8f9" icon={faTrophy} /></div>
          <div className="iconBox"><FontAwesomeIcon color="#f7f8f9" icon={faUsers} /></div>
          <img alt="" src={yo} width={28} height={28} className="imageProfile"/>
        </div>
        <MainClass />
      </div>
    </Provider>
  );
}

export default App;
