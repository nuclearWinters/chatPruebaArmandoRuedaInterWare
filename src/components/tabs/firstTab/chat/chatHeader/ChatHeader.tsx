import React, { Component } from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faHome, faSearch, faStar, faCommentDots, faLink, faFile } from '@fortawesome/free-solid-svg-icons'
import him from "../../../../../imgs/ANGELA.jpg"
import "./ChatHeader.css"
import { toggle } from "../../../../../actions/toggleMenuActions"
import { connect } from 'react-redux'

type PropsChatHeader = {
    toggle: Function
}

class ChatHeader extends Component<PropsChatHeader, {}> {
    render() {
        return (
            <div className="chatHeader">
                <img alt="" className="image" src={him} width={36} height={36} />
                <div className="profileName" >
                    <div>
                        Belle Delphine
                    </div>
                    <div className="profileStatus" >
                        <FontAwesomeIcon color="#797979" icon={faStar} size="xs" className="icon"/>
                        <div className="circle"></div>
                        <div className="chatFont status">Online</div>
                        <FontAwesomeIcon color="#797979" icon={faHome} size="xs"/>
                        <div className="chatFont option">
                            <div className="entry">
                                <div className="entry__inner">
                                      <span className="entry__text">{` \(★ω★)/ YOU  (*≧ω≦*) ARE  ☆ ～('▽^人)  THE  ヽ(>∀<☆)ノ  BEST  (╯✧▽✧)╯`}</span>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
                <div className="inputChatHeader">
                    <input className="chatInput" placeholder="Search lewd polaroid"/>
                    <FontAwesomeIcon color="#797979" icon={faSearch} className="iconSearch" />
                </div>
                <div className="inputChatHeaderBoxFirst">
                    <div className="inputChatHeaderBoxSelected">
                        <FontAwesomeIcon color="#49a9ea" icon={faCommentDots} className="iconSelected"/>
                        <div className="chatFont selected">Chat</div>
                    </div>
                </div>
                <div className="inputChatHeaderBox">
                    <FontAwesomeIcon color="#797979" icon={faFile} className="iconChatHeader"/>
                    <div className="chatFont">Archivos</div>
                </div>
                <div className="inputChatHeaderBoxLast">
                    <FontAwesomeIcon color="#797979" icon={faLink} className="iconChatHeader"/>
                    <div className="chatFont">Links</div>
                </div>
                <div className="burgerIcon" onClick={() => this.props.toggle()}>
                    <div></div>
                    <div></div>
                    <div></div>
                </div>
            </div>
        )
    }
}

export default connect(null, { toggle })(ChatHeader)