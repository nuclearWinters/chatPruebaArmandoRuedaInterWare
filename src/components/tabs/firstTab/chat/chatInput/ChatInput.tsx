import React, { Component } from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faPaperclip, faGrinAlt, faPaperPlane } from '@fortawesome/free-solid-svg-icons'
import yo from "../../../../../imgs/CARLOS.png"
import "./ChatInput.css"
import { connect } from 'react-redux'
import { createPost } from "../../../../../actions/postActions"

type StateChatInput = {
    newMessage: string
}

type PropsChatInput = {
    createPost: Function
}

class ChatInput extends Component<PropsChatInput, StateChatInput> {
    constructor(props: PropsChatInput) {
        super(props);
        this.onChange = this.onChange.bind(this)
        this.onSubmit = this.onSubmit.bind(this)
        this.state = {
            newMessage: ""
        }
    }

    onSubmit = (e: React.KeyboardEvent<HTMLInputElement>) => {
        if (e.key === 'Enter' && this.state.newMessage !== "") {
            this.props.createPost(this.state.newMessage)
            this.setState({
                newMessage: ""
            })
        }
    }

    onChange = (e: React.FormEvent<HTMLInputElement>) => {
        this.setState({newMessage: e.currentTarget.value})
    }

    render() {
        return (
            <div className="chatInputFixed" >
                <img alt="" className="imageChatInputFixed" src={yo} width={36} height={36} />
                <div className="inputOfChatFixedBox" >
                    <input value={this.state.newMessage} type="text" onKeyDown={this.onSubmit} onChange={this.onChange} className="inputOfChatFixed" placeholder="Escribe un comentario"/>
                </div>
                <FontAwesomeIcon color="#797979" icon={faPaperclip} className="iconChatInputFixedFirst"/>
                <FontAwesomeIcon color="#797979" icon={faGrinAlt} className="iconChatInputFixed"/>
                <FontAwesomeIcon color="#797979" icon={faPaperPlane} className="iconChatInputFixedLast"/>
            </div>
        )
    }
}

export default connect(null, { createPost })(ChatInput)