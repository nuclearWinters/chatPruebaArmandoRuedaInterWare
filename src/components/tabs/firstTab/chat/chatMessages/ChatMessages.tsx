import React, { Component } from 'react'
import yo from "../../../../../imgs/CARLOS.png"
import him from "../../../../../imgs/ANGELA.jpg"
import "./ChatMessages.css"
import moment from "moment"

type MessageChat = {
    user: {
        name: string,
        username: string,
        id: number
    },
    receiver_id: number, 
    message: string,
    date?: Date
}

type PropsChat = {
    posts: MessageChat[]
}

export default class ChatMessages extends Component<PropsChat, {}> {
    constructor(props: PropsChat) {
        super(props)
        this.messageEnd = null;
    }
    public messageEnd: HTMLDivElement | null;
    scrollToBottom = () => {
        if (this.messageEnd) this.messageEnd.scrollIntoView(false);
    }
    componentDidUpdate() {
        this.scrollToBottom()
    }
    render() {
        return (
            <div className="chatMessages" >
                <div className="chatMessagePaddingBox">
                    {this.props.posts.map((res: MessageChat, i: number) => 
                        <div key={i} className="messageBox" >
                            {i === 6 ?
                                <div className="divider">
                                    <div className="dividerLine"></div>
                                    <div className="dividerDate">{moment().locale("es").format('dddd, D [de] MMMM')}</div>
                                </div>
                                
                            : null}
                            <div className="messageData" >
                                <img alt="" className="messageImage" src={res.user.id === 1 ? yo : him} width={28} height={28}/>
                                <div className="messageUser">{res.user.name === "Bender" ? "#1 Loser" : "Belle Delphine"}</div>
                                {res.date ? <div className="messageDate">{moment().format('D MMM - h:mm A')}</div>  : <div className="messageDate">5 Nov - 1:37 PM</div>}
                            </div>
                            <div className="messageMessage" >{res.message}</div>
                        </div>
                    )}
                    <div style={{ float:"left", clear: "both" }}
                        ref={el => { this.messageEnd = el; }}>
                    </div>
                </div>
            </div>
        )
    }
}
