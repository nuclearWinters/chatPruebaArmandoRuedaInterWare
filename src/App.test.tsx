import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';
import ChatMessages from './components/tabs/firstTab/chat/chatMessages/ChatMessages';
import ChatHeader from './components/tabs/firstTab/chat/chatHeader/ChatHeader';
import ChatInput from './components/tabs/firstTab/chat/chatInput/ChatInput';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});