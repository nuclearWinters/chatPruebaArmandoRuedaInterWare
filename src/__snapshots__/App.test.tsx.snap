// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatMessages inital six messages not change with fetch 1`] = `
<div
  className="chatMessages"
>
  <div
    className="chatMessagePaddingBox"
  >
    <div
      className="messageBox"
    >
      <div
        className="messageData"
      >
        <img
          alt=""
          className="messageImage"
          height={28}
          src="CARLOS.png"
          width={28}
        />
        <div
          className="messageUser"
        >
          #1 Loser
        </div>
        <div
          className="messageDate"
        >
          5 Nov - 1:37 PM
        </div>
      </div>
      <div
        className="messageMessage"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      </div>
    </div>
    <div
      className="messageBox"
    >
      <div
        className="messageData"
      >
        <img
          alt=""
          className="messageImage"
          height={28}
          src="ANGELA.jpg"
          width={28}
        />
        <div
          className="messageUser"
        >
          Belle Delphine
        </div>
        <div
          className="messageDate"
        >
          5 Nov - 1:37 PM
        </div>
      </div>
      <div
        className="messageMessage"
      >
        Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
      </div>
    </div>
    <div
      className="messageBox"
    >
      <div
        className="messageData"
      >
        <img
          alt=""
          className="messageImage"
          height={28}
          src="CARLOS.png"
          width={28}
        />
        <div
          className="messageUser"
        >
          #1 Loser
        </div>
        <div
          className="messageDate"
        >
          5 Nov - 1:37 PM
        </div>
      </div>
      <div
        className="messageMessage"
      >
        Consectetur adipiscing elit.
      </div>
    </div>
    <div
      className="messageBox"
    >
      <div
        className="messageData"
      >
        <img
          alt=""
          className="messageImage"
          height={28}
          src="CARLOS.png"
          width={28}
        />
        <div
          className="messageUser"
        >
          #1 Loser
        </div>
        <div
          className="messageDate"
        >
          5 Nov - 1:37 PM
        </div>
      </div>
      <div
        className="messageMessage"
      >
        Incididunt ut labore et dolore magna.
      </div>
    </div>
    <div
      className="messageBox"
    >
      <div
        className="messageData"
      >
        <img
          alt=""
          className="messageImage"
          height={28}
          src="ANGELA.jpg"
          width={28}
        />
        <div
          className="messageUser"
        >
          Belle Delphine
        </div>
        <div
          className="messageDate"
        >
          5 Nov - 1:37 PM
        </div>
      </div>
      <div
        className="messageMessage"
      >
        Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
      </div>
    </div>
    <div
      className="messageBox"
    >
      <div
        className="messageData"
      >
        <img
          alt=""
          className="messageImage"
          height={28}
          src="ANGELA.jpg"
          width={28}
        />
        <div
          className="messageUser"
        >
          Belle Delphine
        </div>
        <div
          className="messageDate"
        >
          5 Nov - 1:37 PM
        </div>
      </div>
      <div
        className="messageMessage"
      >
        Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
      </div>
    </div>
    <div
      style={
        Object {
          "clear": "both",
          "float": "left",
        }
      }
    />
  </div>
</div>
`;
